# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/WebCodecsUtils.h
# Commit: b49ae6bc356a
# Full Hash: b49ae6bc356a40be81ed2b18842461cc47abf5cf
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Guess pixel format from VideoData r=aosmond
#   
#   Depends on D170635
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171166
# ==============================================================================

diff -r 0899b9a2539e -r b49ae6bc356a dom/media/webcodecs/WebCodecsUtils.h
--- a/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 22:33:13 2023 +0000
+++ b/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 22:33:13 2023 +0000
@@ -21,6 +21,7 @@
 namespace gfx {
 enum class ColorRange : uint8_t;
 enum class ColorSpace2 : uint8_t;
+enum class SurfaceFormat : int8_t;
 enum class TransferFunction : uint8_t;
 enum class YUVColorSpace : uint8_t;
 }  // namespace gfx
@@ -79,6 +80,20 @@
 
 Maybe<VideoColorPrimaries> ToPrimaries(const gfx::ColorSpace2& aColorSpace);
 
+/*
+ * The following are utilities to convert from gfx's formats to
+ * VideoPixelFormats.
+ */
+
+enum class ImageBitmapFormat : uint8_t;
+enum class VideoPixelFormat : uint8_t;
+
+Maybe<VideoPixelFormat> SurfaceFormatToVideoPixelFormat(
+    gfx::SurfaceFormat aFormat);
+
+Maybe<VideoPixelFormat> ImageBitmapFormatToVideoPixelFormat(
+    ImageBitmapFormat aFormat);
+
 }  // namespace dom
 }  // namespace mozilla
 
